model/riscv_fdext_regs.sail

```diff
@@ -270,30 +270,48 @@ function wF_bits(i: bits(5), data: flenbits) -> unit = {
 
 overload F = {rF_bits, wF_bits, rF, wF}
 
+val rF_H : bits(5) -> bits(16) effect {escape, rreg}
+function rF_H(i) = {
+  assert(sizeof(flen) >= 16);
+  nan_unbox(F(i))
+}
+
 val rF_or_X_H : bits(5) -> bits(16) effect {escape, rreg}
 function rF_or_X_H(i) = {
   assert(sizeof(flen) >= 16);
   assert(sys_enable_fdext() != sys_enable_zfinx());
   if   sys_enable_fdext()
-  then nan_unbox(F(i))
+  then rF_H(i)
```

Comment:
Hoist and use the overloads? The underlying accessors are kind of ugly to use, especially for writes.

Reply:
Given that rF_H() needs to assert sizeof(flen) anyway, I would go as far as rewriting this as (consider pseudocode & ignore the indentation):
```
assert(sys_enable_fdext() != sys_enable_zfinx());
if sys_enable_fdext()
then F_H(i)
else 
  {
     assert(sizeof(flen) >= 16);
     X(i)[15..0]
  }
```

Agreed?

Reply:
What does that achieve? It makes it look like the assertion doesn't need to be true for F_H...

Reply:
Although, does FLEN actually have a meaning for Zfinx? Arguably it's a bit dodgy to query for Zfinx...

Reply:
The same assertion lives inside of F_H(), but we can keep it explicit.  It's a matter of taste and I don't care if we have it twice of only once along each path.  So, let's keep it where it is.

That only leave the use of the overloaded F_H() instead of the direct use of rF_H and wF_H.  Personally, I would use the direct accessor rF_H/wF_H on the rF_or_X_H/wF_or_X_H accessors to keep things explicit — then again, it doesn't make a difference.

So to be clear: the requested change is to use the overload instead of the underlying accessor?

Reply:
Yes; just use F_H rather than [rw]F_H (and therefore move the relevant definitions so they're visible)

Reply:
I think the problem actually is just that it’s poorly-named and should be maxflen, given how it’s defined and used